include/libtorrent/add_torrent_params.hpp

```diff
@@ -313,7 +315,7 @@ namespace libtorrent
 		// can be set to control the initial file priorities when adding a
 		// torrent. The semantics are the same as for
 		// ``torrent_handle::prioritize_files()``.
-		std::vector<std::uint8_t> file_priorities;
+		vector<std::uint8_t, file_index_t> file_priorities;
```

Comment:
quite a change... This will greatly impact my `swig+java` binding, because the type will be translated as another class, and instead of passing primitives, I will be creating  and passing objects just for indexing (super slow).